actionpack/lib/action_controller/metal/implicit_render.rb

```diff
@@ -1,29 +1,48 @@
+require 'active_support/core_ext/string/strip'
+
 module ActionController
+  # Handles implicit rendering for a controller action when it did not
+  # explicitly indicate a response via e.g. +render+, +respond_to+,
+  # +redirect+ or +head+.
+  #
+  # For API controllers, the implicit render always renders
+  # "204 No Content" and does not account for any templates.
+  #
+  # For other controllers, the implicit render still falls back to rendering
+  # "204 No Content", though not before exhausting 3 other options first.
+  #
+  # First, if a template exists for the controller action, it is rendered.
+  # This template lookup takes into account the action name, format, locales,
+  # handlers, variants etc.
+  #
+  # Second, if any templates exist for the controller action in any other
+  # format, variant, etc. an <tt>ActionController::UnknownFormat</tt> is raised
+  # because the available templates is assumed exhaustive. E.g. writing only an
+  # HTML template means only that format is known.
+  #
+  # Third and last, if the current request is a real "interactive" browser request,
+  # <tt>ActionView::MissingTemplate</tt> is raised to display a helpful error
+  # message.
```

Comment:
Some of the docs felt a bit hard to read to me. Took a stab at rephrasing them.

Something feels funny about `assumed exhaustive` too, but ¯\_(ツ)_/¯.

Here's the originals:

```ruby
# Handles the implicit behavior for a controller action when it did not
# explicitly call +render+ or otherwise indicate what the response  should be,
# such as by calling +respond_to+, +redirect+ and +head+.
#
# If this controller action has any associated templates, it tries to render
# the most appropiate template for the current action, taking into account of
# the action name, format, locales, variants etc.
#
# The list of available templates is assumed to be exhausative – if a suitable
# template cannot be found (for example, when the client requested the XML
# format when the only templates in HTML and JSON formats are available), an
# <tt>ActionController::UnknownFormat</tt> error will be raised.
#
# On the other hand, when there are no templates associated with the controller
# action at all, it tries to determine if the current request is "interactive"
# (i.e. came from a real browser). If so, it raises the <tt>ActionView::MissingTemplate</tt>
# error in order to display a helpful error message. Otherwise, it assumes the
# request is an API request and renders a "204 No Content" response.
#
# For API controllers, the implicit render is always "204 No Content" without
# performing any template lookups.
```